System/Posix/Fcntl.hsc

```diff
@@ -87,8 +87,16 @@ fileAdvise _ _ _ _ = return ()
 -- @since 2.7.1.0
 fileAllocate :: Fd -> FileOffset -> FileOffset -> IO ()
 #if HAVE_POSIX_FALLOCATE
+#if defined(freebsd_HOST_OS)
 fileAllocate fd off len = do
   throwErrnoIfMinus1_ "fileAllocate" (c_posix_fallocate (fromIntegral fd) (fromIntegral off) (fromIntegral len))
+#else
```

Comment:
The FreeBSD exception is obsolete as of FreeBSD 11.0. Just go with the same code on all platforms that have `posix_fallocate(2)`.

Reply:
Done.